### PR TITLE
feat: [SRM-13812]: ELK Logs implementation on Common Health source framework.

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
 ELK Logs implementation on Common Health source framework.

#### Screenshots
New Note

New Note

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  loading,
+  queryFieldIdentifier,
+  values
+}: {
+  query: string
+  loading: boolean
+  queryFieldIdentifier: string | undefined
+  values: CommonCustomMetricFormikInterface
+}): boolean {
+  return Boolean(isEmpty(query) || loading || getIsQueryFieldNotPresent(queryFieldIdentifier, values))
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:15]: Boolean wrapper can be avoided as 
isEmpty(query) is Boolean 
loading is Boolean 
getIsQueryFieldNotPresent(queryFieldIdentifier, values) is Boolean

Comment (harshilbits) [reply src/modules/85-cv/components/CommonQueryViewer/CommonQueryViewer.utils.ts:15]: No harm in keeping in ensuring the type safety of return type.

File: src/modules/85-cv/pages/health-source/common/CommonCustomMetric/CommonCustomMetric.tsx [hunk 2]
```diff
@@ -75,12 +75,12 @@ export default function CommonCustomMetric(props: CommonCustomMetricInterface):
...
     let isUpdated = false
-    const isServiceInstanceFixed = getMultiTypeFromValue(formikValues.serviceInstance) === MultiTypeInputType.FIXED
+    const isServiceInstanceFixed = getMultiTypeFromValue(formikValues.serviceInstanceField) === MultiTypeInputType.FIXED
 
     if (isQueryRuntimeOrExpression) {
-      const canMakeRuntime = !formikValues.serviceInstance || isServiceInstanceFixed
+      const canMakeRuntime = !formikValues.serviceInstanceField || isServiceInstanceFixed
       if (canMakeRuntime) {
-        formikValues.serviceInstance = RUNTIME_INPUT_VALUE
+        formikValues.serviceInstanceField = RUNTIME_INPUT_VALUE
```
[the remark this answers is not in the document]
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/common/CommonCustomMetric/CommonCustomMetric.tsx:83]: This is the old change done by you I think , as part of this PR i have just updated the field `serviceInstance ` to `serviceInstanceField `

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.tsx [hunk 3]
```diff
@@ -47,11 +49,14 @@ export default function CommonCustomMetricFormContainer(props: CommonCustomMetri
   const { isQueryRuntimeOrExpression } = useCommonHealthSource()
   const { projectIdentifier, orgIdentifier, accountId } = useParams<ProjectPathProps>()
   const chartConfig = healthSourceConfig?.customMetrics?.metricsChart
+  const queryField = healthSourceConfig.customMetrics?.queryAndRecords?.queryField
+  const queryFieldValue = (queryField ? values[queryField.identifier] : '') as string
   const providerType = HEALTHSOURCE_TYPE_TO_PROVIDER_MAPPING[product?.value || sourceType]
+  // TODO - this will not be needed once the backend refactoring for enumns is done.
```
Comment (praneshg239) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.tsx:55]: Do we still need this?

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.tsx (hunk 3 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.tsx:55]: Yes , this is required . Backend changes are still not merged.

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts [hunk 4]
```diff
@@ -598,3 +624,22 @@ export function getCurrentQueryData(
       : {}
   ) as CommonCustomMetricFormikInterface
 }
+
+export function getFieldName(fieldIdentifier: keyof CommonCustomMetricFormikInterface): string {
+  switch (fieldIdentifier) {
+    case CustomMetricFormFieldNames.QUERY:
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts:630]: these are for error messages , right ?  Should we be picking them from strings ?

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/HealthSourceConfigs/HealthSourceConfigs.ts [hunk 5]
```diff
@@ -79,5 +81,68 @@ export const healthSourcesConfig: HealthSourcesConfig = {
...
+    customMetrics: {
+      enabled: true,
+      queryAndRecords: {
+        enabled: true,
+        titleStringKey: 'cv.monitoringSources.commonHealthSource.defineQueryDescriptionMetrics',
+        queryField: {
+          type: FIELD_ENUM.DROPDOWN,
+          label: 'Log Indexes',
+          identifier: CustomMetricFormFieldNames.INDEX,
+          placeholder: 'Select Log Index',
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/HealthSourceConfigs/HealthSourceConfigs.ts:99]: will we be showing placeholder from config? 
We can pick from string file , right ?

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/components/CommonHealthSourceField/__tests__/CommonHealthSourceField.test.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+      value={{ isTemplate }}
+    >
+      <TestWrapper>
+        <CommonHealthSourceField {...props} />
+      </TestWrapper>
+    </SetupSourceTabsContext.Provider>
+  )
+}
+
+describe('Test cases for CVSLODetailsPage', () => {
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/components/CommonHealthSourceField/__tests__/CommonHealthSourceField.test.tsx:33]: Is this title correct ?

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts (hunk 4 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts:630]: These are label names of the fields.

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/components/CommonHealthSourceField/__tests__/CommonHealthSourceField.test.tsx (hunk 6 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/components/CommonHealthSourceField/__tests__/CommonHealthSourceField.test.tsx:33]: corrcted.

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/HealthSourceConfigs/HealthSourceConfigs.ts (hunk 5 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/HealthSourceConfigs/HealthSourceConfigs.ts:99]: Keeping in config file can provide more flexibility and UI can be more config driven than keeping things tied to the codebase.

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts (hunk 4 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts:637]: Added them in strings.

**Review by deepesh-ui — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

**Pull request merged**